### PR TITLE
qt6 port

### PR DESCRIPTION
**Pull request by hsitter**
sits on top of https://github.com/flathub/org.kde.spectacle/pull/9

### CODE REVIEW

File: org.kde.spectacle.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
         "--filesystem=xdg-pictures",
         "--talk-name=org.kde.kglobalaccel",
         "--talk-name=org.kde.plasmashell",
-        "--talk-name=org.kde.kwin",
+        "--talk-name=org.kde.KWin",
```
Comment (travier) [inline org.kde.spectacle.json:16]: This is weird. Was the name incorrect or was it get changed?

Comment (hsitter) [reply org.kde.spectacle.json:16]: was incorrect

**Comment by travier** [comment]
```
{
    "errors": [
        "appstream-failed-validation",
        "appstream-missing-developer-name"
    ],
    "appstream": [
        "E: org.kde.spectacle.desktop:~: desktop-app-launchable-omitted"
    ],
    "message": "Please consult the documentation at https://docs.flathub.org/docs/for-app-authors/linter"
}
```

**Comment by hsitter** [comment]
I've had a look at wayland and I am honestly not sure spectacle as a flatpak is worth the trouble. Practically it only works with kwin and practically it is shipped by every distro since it is part of core workspace functionality.

Specifically we need dbus access for screenshots and wayland protocol access for screen casting.

The former is hugely complicated by security hardening inside kwin and the fact that dbus communication happens through the proxy (so kwin doesn't actually know which executable they are talking to). That is technically solvable but at relatively large engineering cost (we basically have to map our way backwards from the proxy -> cgroup -> flatpak id -> flatpak instance).

The latter is currently impossible because AFAICS flatpak doesn't implement https://wayland.app/protocols/security-context-v1#wp_security_context_manager_v1 and so kwin has no way of knowing which service is requesting the wayland permission to record the screen.

**Comment by travier** [comment]
I agree. This is still in beta because of all those limitations.

**Comment by travier** [comment]
```
{
    "errors": [
        "appstream-id-mismatch-flatpak-id"
    ],
    "message": "Please consult the documentation at https://docs.flathub.org/docs/for-app-authors/linter"
}
```
🤔

**Comment by bbhtt** [comment]
bot, build

**Pull request merged**